metadata-io/src/main/java/com/linkedin/metadata/graph/neo4j/Neo4jGraphService.java

```diff
@@ -524,4 +747,19 @@ private Statement getOrInsertNode(@Nonnull Urn urn) {
   public boolean supportsMultiHop() {
     return true;
   }
+
+  private boolean getRealDirection(@Nonnull String sourceType, @Nonnull String relationshipType) {
+    // Get real direction by check INCOMING/OUTGOING direction and RelationshipType
+    LineageRegistry.LineageSpec sourceLineageSpec = getLineageRegistry().getLineageSpec(sourceType);
+    if (sourceLineageSpec != null) {
+      List<LineageRegistry.EdgeInfo> upstreamCheck = sourceLineageSpec.getUpstreamEdges()
+          .stream()
+          .filter(t -> t.getDirection() == RelationshipDirection.OUTGOING && t.getType().equals(relationshipType))
+          .collect(Collectors.toList());
+      if (upstreamCheck.size() > 0 || sourceType.equals("schemaField")) {
```

Comment:
nit: we can say `if (upstreamCheck.isEmpty() || sourceType.equals("schemaField"))`

Reply:
changed all .size() > 0 to .isEmpty()